snapcraft/internal/errors.py

```diff
@@ -39,6 +39,14 @@ class MissingState(Exception):
     pass
 
 
+class FileConflictError(SnapcraftError):
+
+    fmt = (
+        'The following files would create a conflict in the snap: '
+        '{fileset!r}'
```

Comment:
This is still pretty generic. However, is this exception ever raised except in this very specific context? Must we keep it generic? Something like "The following files have been excluded by the 'stage' keyword, but included by the 'prime' keyword: {fileset!r}" tells the user how to solve the problem.